src/libsync/discoveryphase.h

```diff
@@ -182,13 +189,21 @@ private slots:
     // If this directory is an external storage (The first item has 'M' in its permission)
     bool _isExternalStorage = false;
     // If this directory is e2ee
-    SyncFileItem::EncryptionStatus _isE2eEncrypted = SyncFileItem::EncryptionStatus::NotEncrypted;
+    SyncFileItem::EncryptionStatus _encryptionStatusCurrent = SyncFileItem::EncryptionStatus::NotEncrypted;
     bool _isFileDropDetected = false;
     bool _encryptedMetadataNeedUpdate = false;
+    SyncFileItem::EncryptionStatus _encryptionStatusRequired = SyncFileItem::EncryptionStatus::NotEncrypted;
     // If set, the discovery will finish with an error
     int64_t _size = 0;
     QString _error;
     QPointer<LsColJob> _lsColJob;
+    
+    QSharedPointer<FolderMetadata> _e2EeFolderMetadata;
+    QSharedPointer<FolderMetadata> _topLevelE2eeFolderMetadata;
+
+    QSet<QString> _listRootE2eeFolders;
```

Comment:
I do not get why you need this here
are you doing discovery action in many root e2ee folder from a single one ?
I do not get why it cannot be told what its root encrypted folder is

Reply:
@mgallien I have removed some leftovers, and now only `_topLevelE2eeFolderPaths` is left, it is required as we need to fetch the top-level folder's metadata from inside the nested folder's metadata, as the data for decryption is only stored in top-level folders as per E2EE V2. See comments, and check comments I've added